server/athenian/api/controllers/miners/github/contributors.py

```diff
@@ -21,14 +21,42 @@ async def mine_contributors(repos: Collection[str],
                             time_to: Optional[datetime],
                             db: databases.Database,
                             cache: Optional[aiomcache.Client],
-                            with_stats: Optional[bool] = True) -> List[dict]:
+                            with_stats: Optional[bool] = True,
+                            with_roles: List[str] = None) -> List[dict]:
     """Discover developers who made any important action in the given repositories and \
     in the given time frame."""
     time_from = time_from or datetime(1970, 1, 1, tzinfo=timezone.utc)
     now = datetime.now(timezone.utc) + timedelta(days=1)
     time_to = time_to or datetime(now.year, now.month, now.day, tzinfo=timezone.utc)
 
-    return await _mine_contributors(repos, time_from, time_to, with_stats, db, cache)
+    if with_roles:
+        fetch_stats = True
```

Comment:
I think we should split the function into two at this point, the logic is getting complex.

1. The old one that is used in `/contributors`
2. The new one with blows and whistles.

Some code will be duplicated, and this is fine.

Reply:
I'd ask @warenlg's input here because I seem to recall that he recently mentioned at some point we will need to also filter/sort the contributors as well when creating/updating a team.

Reply:
Yes we need one stat to sort the users listed in the breakdown of the team set up page https://athenianco.atlassian.net/browse/DEV-112

Reply:
Okay so @vmarkovtsev I'll keep the functionality in common and I'll refactor it according to what you said [here](https://github.com/athenianco/athenian-api/pull/447#discussion_r437198072).